### PR TITLE
ci: build docker image

### LINKED ISSUE #25: Build docker image on the CI (release)
The title is self-descriptive

### PR DESCRIPTION
**Pull request by joaopereira12**
Resolves #25 

I've added a job to the Build Packages workflow that builds the docker images with docker compose.

### CODE REVIEW

**Comment by d-roak** [comment]
It's not this. It's building the node docker image, and uploading it to the gh registry. It's not related w/ docker compose

**Comment by joaopereira12** [comment]
> It's not this. It's building the node docker image, and uploading it to the gh registry. It's not related w/ docker compose

Okay, my bad, i though the image could be buildt with compose, I will fix it tomorrow.

**Comment by joaopereira12** [comment]
I tested it on my forked project and the docker image was registered as you can see in the image bellow.

image

File: .github/workflows/build-docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Build Docker Image
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```
Comment (d-roak) [inline .github/workflows/build-docker-image.yml:6]: release

File: .github/workflows/build-docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions: 
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
```
Comment (d-roak) [inline .github/workflows/build-docker-image.yml:19]: do we need all of these?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -58,3 +58,5 @@ jobs:
           cd ../../examples/canvas
           yarn install --frozen-lockfile
           yarn build
+
+
```
Comment (d-roak) [inline .github/workflows/build.yml:62]: ??

File: .github/workflows/build-docker-image.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
+        with:
+          context: ./packages/node
+          file: ./packages/node/Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+
+
```
Comment (d-roak) [inline .github/workflows/build-docker-image.yml:46]: don't leave empty spaces at the end of files

File: .github/workflows/build-docker-image.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
```
Comment (d-roak) [inline .github/workflows/build-docker-image.yml:25]: aren't there versions for the actions (same applies for metadata-action and build-push-action)?

File: .github/workflows/build-docker-image.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+      - name: Build and push Docker image
+        id: push
```
Comment (d-roak) [inline .github/workflows/build-docker-image.yml:36]: why do we need this id?

**Review by d-roak — CHANGES_REQUESTED** [review]

**Comment by joaopereira12** [comment]
I've fixed the file and added multi architectures build (arm & amd).

**Review by d-roak — APPROVED** [review]
lgtm

**Pull request merged**